File: exemplos_python/Paredes.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Paredes

Percorre o campo de batalha encostado às paredes no sentido dos ponteiros do relógio.
"""
from port_module import Robot
from time import sleep

class Paredes( Robot ):
    def __init__( self ):
        Robot.__init__( self )
        self._alive = True
        # Inicia o robot
        self.init()



    def round_started( self, roundNum ):
        print "round_started( " + str( roundNum ) + ' )'
        self.subscribe_events( "[out_of_energy]" )
        


    def out_of_energy( self ):
        """
        Quando fica sem energia espera um segundo para recarregar e reduz a velocidade para reduzir o consumo.
        """
        print "out_of_energy()"
        # Faz uma pausa para recarregar as baterias ate conseguir andar novamente
        sleep( 1 )
        # Reduz a velocidade para reduzir o consumo de energia
        self.set_speed( 2 )


    
    def start( self ):
        direction = 0
        while ( self._alive ):
            self.robot_move_forward( 1000 )
            self.robot_turn_to( direction )
            direction += 90
            if ( direction == 360 ): 
                direction = 0

            


if ( __name__ == "__main__" ):
    robot = Paredes()
    robot.start()

File: exemplos_java/Paredes.java
/*
 * Robot: Paredes
 * 
 * Percorre o campo de batalha encostado às paredes no sentido dos ponteiros do relógio.
 */

import java.io.*;
import java.util.*;


class Paredes extends Robot {
	
	public void round_started( int args ) {
		System.out.println( "round_started( " + args +  " )" );
        subscribe_events( "[out_of_energy]" );
	}
	
	
	
	public void out_of_energy() {
		System.out.println( "out_of_energy()" );
        try {
        	Thread.sleep(1000);
        }
        catch(Exception e){}
		set_speed( 2 );
	}

	
	
	
	// Enquanto true o robô continua a funcionar
	boolean alive = true;

	void start() {
		int direction = 0;
		// Inicia o robô
		init();
        while ( alive ) {
            robot_move_forward( 1000 );
            robot_turn_to( direction );
            direction += 90;
            if ( direction == 360 )
                direction = 0;
        }
	}
	
	
	
	
	
	

	public static void main( String[] args ) {
		Paredes robot = new Paredes();
		robot.start();
	}
	
}